components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -38,16 +39,20 @@
 #define MBED_CONF_ESP8266_CTS NC
 #endif
 
+#ifndef MBED_CONF_ESP8266_RST
+#define MBED_CONF_ESP8266_RST NC
+#endif
+
 #define TRACE_GROUP  "ESPI" // ESP8266 Interface
 
 using namespace mbed;
 
 #if defined MBED_CONF_ESP8266_TX && defined MBED_CONF_ESP8266_RX
 ESP8266Interface::ESP8266Interface()
     : _esp(MBED_CONF_ESP8266_TX, MBED_CONF_ESP8266_RX, MBED_CONF_ESP8266_DEBUG, MBED_CONF_ESP8266_RTS, MBED_CONF_ESP8266_CTS),
+      _rst_pin(DigitalOut(MBED_CONF_ESP8266_RST, 1)), // Notice that Pin7 CH_EN cannot be left floating if used as reset
```

Comment:
That initialisation is weird. You're copying a `DigitalOut` object. Surprised that's even permitted - would have expected it to have `NonCopyable` marking.

Hmmm. There is some sort of weird custom assignment operator, so maybe this works, but is pretty odd. Not quite sure how that assignment operator was supposed to be used. Can't you just pass the pin number, and initialise `DigitalOut` normally in `ResetPin`?

Reply:
Fixed